### PR TITLE
Added django-admin command to register profiles for users without one.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
All users who do not have a profile will now have a profile with initial information on profile set to null.

### CODE REVIEW

**Comment by saadyousafarbi** [comment]
@zubair-arbi
Please have a look and review changes.

File: autoshop/core/admin.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
-# -*- coding: utf-8 -*-
-from __future__ import unicode_literals
-
 from django.contrib import admin
+from core.models import Profile
+
+
+class ProfileAdmin(admin.ModelAdmin):
+    list_display = ['user', 'bio', 'gender', 'date_of_birth', 'address', 'mobile_number', 'city', 'country']
```
Comment (zubair-arbi) [inline autoshop/core/admin.py]: on main django admin page we only display the necessary information not all the fields

File: autoshop/core/management/commands/register_profile.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from core.models import Profile
```
Comment (zubair-arbi) [inline autoshop/core/management/commands/register_profile.py]: order imports properly,
1. core
2. third party (including django)
3. local

File: autoshop/core/management/commands/register_profile.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+from django.contrib.auth.models import User
+from django.core.management import BaseCommand
+
+
+class Command(BaseCommand):
+    help = 'Creating profiles for users that dont have a profile.'
```
Comment (zubair-arbi) [inline autoshop/core/management/commands/register_profile.py]: `Create missing user profiles with empty values.`

File: autoshop/core/management/commands/register_profile.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    def handle(self, *args, **options):
+        all_user_ids = User.objects.values_list('id', flat=True)
+        all_profile_ids = Profile.objects.values_list('user_id', flat=True)
+        users_with_unregistered_profiles = list(set(all_user_ids) - set(all_profile_ids))
```
Comment (zubair-arbi) [inline autoshop/core/management/commands/register_profile.py]: can we get all users without profile using query?

File: autoshop/core/management/commands/register_profile.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        if not users_with_unregistered_profiles:
+            self.stdout.write(self.style.SUCCESS('All users have profiles.'))
```
Comment (zubair-arbi) [inline autoshop/core/management/commands/register_profile.py]: add a method to create missing profiles of users`create_profile_for_users(self, users)` and move this logic there.

File: autoshop/core/management/commands/register_profile.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+            return
+        self.stdout.write(self.style.SUCCESS('User id without profile: ' + str(users_with_unregistered_profiles)))
+        for unregistered_profile_id in users_with_unregistered_profiles:
+            Profile.objects.create(user_id=unregistered_profile_id)
+            self.stdout.write(self.style.SUCCESS('Created profile for user: ' + str(unregistered_profile_id)))
```
Comment (zubair-arbi) [inline autoshop/core/management/commands/register_profile.py]: The output I expect should be:
```
Created 3 missing profiles out of total 50 users.
```

File: requirements.txt [hunk 7]
```diff
@@ -1,2 +1,2 @@
-Django==1.4.1
+Django==1.11.1
```
Comment (zubair-arbi) [inline requirements.txt:1]: make sure your local env is also set with `Django==1.11.1`

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Please address this feedback.

File: requirements.txt (hunk 7 above)
Comment (saadyousafarbi) [reply requirements.txt:1]: yes it is.

File: autoshop/core/management/commands/register_profile.py (hunk 4 above)
Comment (saadyousafarbi) [reply autoshop/core/management/commands/register_profile.py]: no

Comment (zubair-arbi) [reply autoshop/core/management/commands/register_profile.py]: Hint: We can!

File: autoshop/core/admin.py [hunk 8]
```diff
@@ -1,6 +1,9 @@
-# -*- coding: utf-8 -*-
-from __future__ import unicode_literals
-
 from django.contrib import admin
+from core.models import Profile
+
+
+class ProfileAdmin(admin.ModelAdmin):
+    list_display = ['user', 'bio', 'gender', 'date_of_birth']
+    fields = ['user']
```
Comment (zubair-arbi) [inline autoshop/core/admin.py]: We need to view all fields for editing in django admin not just the `user`

File: autoshop/core/admin.py [hunk 9]
```diff
@@ -1,6 +1,9 @@
-# -*- coding: utf-8 -*-
-from __future__ import unicode_literals
-
 from django.contrib import admin
+from core.models import Profile
+
+
+class ProfileAdmin(admin.ModelAdmin):
+    list_display = ['user', 'bio', 'gender', 'date_of_birth']
```
Comment (zubair-arbi) [inline autoshop/core/admin.py]: remove bio` for list display as some bio's could be too large to display in list view.

File: autoshop/core/management/commands/register_profile.py (hunk 4 above)
Comment (zubair-arbi) [reply autoshop/core/management/commands/register_profile.py]:
```sql
>>> User.objects.filter(profile=None)
[<User: ecommerce_worker>, <User: admin>, <User: course_discovery_worker>, <User: enterprise_worker>]
>>> q=User.objects.filter(profile=None)
>>> print q.query
SELECT `auth_user`.`id`, `auth_user`.`password`, `auth_user`.`last_login`, `auth_user`.`is_superuser`, `auth_user`.`username`, `auth_user`.`first_name`, `auth_user`.`last_name`, `auth_user`.`email`, `auth_user`.`is_staff`, `auth_user`.`is_active`, `auth_user`.`date_joined` FROM `auth_user` LEFT OUTER JOIN `auth_userprofile` ON ( `auth_user`.`id` = `auth_userprofile`.`user_id` ) WHERE `auth_userprofile`.`id` IS NULL
```

File: autoshop/core/views.py [hunk 10]
```diff
@@ -139,16 +138,14 @@ def edit_profile(request):
...
+                mobile_number= form.data['mobile_number'],
                 address = form.data['address'],
-                city = form.data['city'],
-                country = form.data['country'],
+                city= form.data['city'],
+                country= form.data['country'],
             )
             return redirect(reverse('core:profile'))
-
-        print 'Profile for user "%s" failed to save due to validation errors: %s' % (request.user.username, form.errors)
```
Comment (zubair-arbi) [inline autoshop/core/views.py]: Revert this. We will keep it until we have implemented loggers

**Review by zubair-arbi — CHANGES_REQUESTED** [review]

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍

**Pull request merged**